server/auth/types/kerberos/kerberos_auth.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import {
+  CoreSetup,
+  SessionStorageFactory,
+  IRouter,
+  ILegacyClusterClient,
+  OpenSearchDashboardsRequest,
+  Logger,
+  LifecycleResponseFactory,
+  AuthToolkit,
+} from 'opensearch-dashboards/server';
+import { OpenSearchDashboardsResponse } from 'src/core/server/http/router';
+import { SecurityPluginConfigType } from '../../..';
+import { SecuritySessionCookie } from '../../../session/security_cookie';
+import { KerberosAuthRoutes } from './routes';
+import { AuthenticationType } from '../authentication_type';
+import { composeNextUrlQueryParam } from '../../../utils/next_url';
+import { KERBEROS_AUTH_LOGIN } from '../../../../common';
+
+export class KerberosAuthentication extends AuthenticationType {
+  public readonly jwtCookie: string = 'jwt';
+  public readonly kerberosHeader: string = 'Negotiate';
+
+  constructor(
+    config: SecurityPluginConfigType,
+    sessionStorageFactory: SessionStorageFactory<SecuritySessionCookie>,
+    router: IRouter,
+    esClient: ILegacyClusterClient,
```

Comment:
Can we use IClusterClient as ILegacyClusterClient is [marked as deprecated](https://github.com/DarshitChanpura/OpenSearch-Dashboards/blob/main/src/core/server/opensearch/legacy/cluster_client.ts#L108).